connect/mirror/src/test/java/org/apache/kafka/connect/mirror/MirrorSourceTaskTest.java

```diff
@@ -256,39 +277,76 @@ public void testPartitionStateMutation() {
         partitionStates.put(sourceTopicPartition, partitionState);
         RecordMetadata recordMetadata = new RecordMetadata(sourceTopicPartition, metadataOffset, 0, 0, 0, recordPartition);
 
-        when(outstandingOffsetSyncs.tryAcquire()).thenReturn(true);
+        doAnswer(new Answer<Object>() {
+            @Override
+            public Object answer(final InvocationOnMock invocation) {
+                final Callback callback = invocation.getArgument(1);
+                callback.onCompletion(null, null);
+                return null;
+            }
+        }).when(producer).send(any(), any());
```

Comment:
This can be simplified a bit by using an `ArgumentCaptor`:

```suggestion
        ArgumentCaptor<Callback> producerCallback = ArgumentCaptor.forClass(Callback.class);
        when(producer.send(any(), producerCallback.capture())).thenAnswer(mockInvocation -> {
            producerCallback.getValue().onCompletion(null, null);
            return null;
        });
```

Reply:
Updated.